src/Plugin.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * The main plugin file.
+ *
+ * If we evolve from a canonical plugin into WordPress core, this file would be left behind.
+ *
+ * @package WP\MCP
+ */
+
+declare(strict_types = 1);
+
+namespace WP\MCP;
+
+use WP\MCP\Core\McpAdapter;
+
+/**
+ * Class - Plugin
+ */
+final class Plugin {
+	/**
+	 * The one true plugin.
+	 *
+	 * @var ?static
+	 */
+	protected static $instance;
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public static function instance(): self {
+		if ( ! isset( self::$instance ) ) {
+			self::$instance = new self();
+			self::$instance->setup();
+
+			/**
+			 * Fires after the main plugin class has been initialized.
+			 *
+			 * @param self $instance The main plugin class instance.
+			 */
+			do_action( 'wp_mcp_init', self::$instance );
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Setup the plugin.
+	 */
+	private function setup(): void {
+		McpAdapter::instance();
```

Comment:
This is more illustrative, since im pretty sure `McpAdapter` is a singleton and doesn't need to be instantiated before use.
If in theory there was future MCP-cetric stuff that did need centralized loading, this would be the place.

Reply:
Done in #18